### PR TITLE
random and wiggle rotations Patch 1

### PR DESCRIPTION
**Pull request by og76**
random eul / quat + wiggle  eul / quat
(just like the vectors)
for quat I keep
- w = 1 in random, cause otherwise it's very hard to control and scale does not matter
this way it works only for rotation, just like the eul variant
- w = amplitude.w same reasons. still, here I give control over w so that you can tweak it other than 1, but not the randomness

### CODE REVIEW

File: nodes/rotation/wiggle_euler.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class EulerWiggleNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_EulerWiggleNode"
+    bl_label = "Euler Wiggle"
+
+    nodeSeed = IntProperty(update = propertyChanged)
+
+    def create(self):
+        self.inputs.new("an_FloatSocket", "Seed", "seed")
+        self.inputs.new("an_FloatSocket", "Evolution", "evolution")
+        self.inputs.new("an_EulerSocket", "Amplitude", "amplitude").value = [1.570796, 1.570796, 1.570796]
```
Comment (JacquesLucke) [inline nodes/rotation/wiggle_euler.py]: would be better to say math.pi / 2 instead of this 'magic number'

File: nodes/rotation/random_euler.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class RandomEulerNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_RandomEulerNode"
+    bl_label = "Random Euler"
+
+    nodeSeed = IntProperty(name = "Node Seed", update = propertyChanged, max = 1000, min = 0)
+
+    def create(self):
+        self.inputs.new("an_IntegerSocket", "Seed", "seed")
+        self.inputs.new("an_FloatSocket", "Scale", "scale").value = 0.8
```
Comment (JacquesLucke) [inline nodes/rotation/random_euler.py]: why 0.8?

**Comment by og76** [comment]
yes, also changed the values to smaller ones, so that you can see what is happening 
(rotations should have smaller val, i.e. for 90* you get to flip from one side to another and the movement seams chaos)
eul to 30* (pi/6) quat to 0.3

File: nodes/rotation/wiggle_euler.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class EulerWiggleNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_EulerWiggleNode"
+    bl_label = "Euler Wiggle"
+
+    nodeSeed = IntProperty(update = propertyChanged)
+
+    def create(self):
+        self.inputs.new("an_FloatSocket", "Seed", "seed")
+        self.inputs.new("an_FloatSocket", "Evolution", "evolution")
+        self.inputs.new("an_EulerSocket", "Amplitude", "amplitude").value = [pi / 6, pi / 6, pi / 6]
```
Comment (JacquesLucke) [inline nodes/rotation/wiggle_euler.py]: sorry, just get the idea. Maybe its even better to use https://docs.python.org/3/library/math.html#math.radians here..

File: nodes/rotation/random_euler.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+class RandomEulerNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_RandomEulerNode"
+    bl_label = "Random Euler"
+
+    nodeSeed = IntProperty(name = "Node Seed", update = propertyChanged, max = 1000, min = 0)
+
+    def create(self):
+        self.inputs.new("an_IntegerSocket", "Seed", "seed")
+        self.inputs.new("an_FloatSocket", "Scale", "scale").value = pi / 6
```
Comment (JacquesLucke) [inline nodes/rotation/random_euler.py]: same here

**Comment by JacquesLucke** [comment]
Thanks

**Pull request merged**